src/main/kotlin/org/rust/lang/core/completion/CompletionEngine.kt

```diff
@@ -72,6 +73,22 @@ object CompletionEngine {
             ?.mapNotNull { it.libTarget }
             ?.map { LookupElementBuilder.create(extCrate, it.normName).withIcon(extCrate.getIcon(0)) }
             ?.toTypedArray() ?: emptyArray()
+
+    fun completeMod(mod: RsModDeclItem): Array<out LookupElement> {
+        val directory = mod.containingMod.ownedDirectory ?: return EMPTY_ARRAY
+
+        val modFromRsFile = directory.files
+            .filter { it.name.endsWith(".rs") }
+            .map { it.name.substringBeforeLast('.') }
+            .filter { it != "mod" }
```

Comment:
I think we could also filter out the name of the module we're in. Something like `it != "mod" && it != mod.parentOfType<PsiFile>()?.name?.substringBeforeLast('.')`.